src/dodal/testing_utils/device_fixtures.py

```diff
@@ -0,0 +1,215 @@
+from functools import partial
+from unittest.mock import MagicMock, patch
+
+import bluesky.plan_stubs as bps
+import pytest
+from ophyd.sim import NullStatus, make_fake_device
+from ophyd_async.core import AsyncStatus, DeviceCollector, set_mock_value
+
+from dodal.devices.aperturescatterguard import (
+    TEST_APERTURE_POSITIONS,
+    AperturePositions,
+    ApertureScatterguard,
+)
+from dodal.devices.aperturescatterguard import get_mock_device as get_mock_ap_sg
+from dodal.devices.backlight import Backlight
+from dodal.devices.eiger import get_mock_device as get_mock_eiger
+from dodal.devices.eiger_odin import EigerOdin
+from dodal.devices.fast_grid_scan import get_mock_device as get_mock_fgs
+from dodal.devices.focusing_mirror import get_mock_voltages
+from dodal.devices.i24.dual_backlight import DualBacklight
+from dodal.devices.oav.oav_detector import get_mock_device as get_mock_oav
+from dodal.devices.robot import get_mock_device as get_mock_bart_robot
+from dodal.devices.slits import Slits
+from dodal.devices.smargon import Smargon
+from dodal.devices.synchrotron import get_mock_device as get_mock_synchrotron
+from dodal.devices.undulator import Undulator
+from dodal.devices.undulator_dcm import get_mock_device as get_mock_undulator_dcm
+from dodal.devices.xspress3_mini.xspress3_mini import Xspress3Mini
+from dodal.devices.zocalo.zocalo_results import ZOCALO_READING_PLAN_NAME, ZocaloResults
+from dodal.testing_utils import constants
+
+from .utility_functions import (
+    create_new_detector_params,
+    patch_ophyd_async_motor,
+    patch_ophyd_motor,
+)
+
+ApSgAndLog = tuple[ApertureScatterguard, MagicMock]
+
+
+@pytest.fixture
+def test_aperture_positions():
+    aperture_positions = AperturePositions.from_gda_beamline_params(
+        TEST_APERTURE_POSITIONS
+    )
+    return aperture_positions
+
+
+@pytest.fixture
+async def mock_aperturescatterguard_with_call_log():
+    call_log = MagicMock()
+    ap_sg = await get_mock_ap_sg()
+    with (
```

Comment:
Should: `patch_ophyd_async_motor` isn't a context manager

Reply:
yes it is, it's type-hinted wrong in ophyd-async, fix has been merged but not released, doesn't affect functionality